src/create-subscription-orders.js

```diff
@@ -0,0 +1,243 @@
+import fetch from 'isomorphic-fetch'
+import pMap from 'p-map'
+import parser from 'cron-parser'
+import { getSubscriptionConfig } from './config.js'
+import { updateOrderWithRetry } from './utils/utils.js'
+import {
+  ACTIVE_STATE,
+  ERROR_STATE,
+  REMINDER_SENT_STATE,
+  SEND_REMINDER_STATE,
+} from './states-constants.js'
+
+let apiRoot
+let ctpClient
+let logger
+let stats
+
+async function createSubscriptionOrders({
+  apiRoot: _apiRoot,
+  ctpClient: _ctpClient,
+  logger: _logger,
+  stateKeyToIdMap,
+}) {
+  stats = {
+    processedTemplateOrders: 0,
+    subscriptionOrderAlreadyCreated: 0,
+    successUrlCalls: 0,
+    recoverableErrorUrlCalls: 0,
+    unrecoverableErrorUrlCalls: 0,
+  }
+  apiRoot = _apiRoot
+  ctpClient = _ctpClient
+  logger = _logger
+  const stateIds = [
+    stateKeyToIdMap.get(ACTIVE_STATE),
+    stateKeyToIdMap.get(SEND_REMINDER_STATE),
+    stateKeyToIdMap.get(REMINDER_SENT_STATE),
+  ]
+  const orderQuery =
+    await _buildQueryForTemplateOrdersThatNeedSubscriptionOrders(stateIds)
+
+  for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
```

Comment:
might be good to encapsulate with try-catch ?

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/10/commits/07eaae4ba25355c40db39e4ef37a9d814b1ed4db

Reply:
I see what you mean, then should we skip for sendReminder commands, wdyt ?